logging/include/Logging.hpp

```diff
@@ -61,15 +74,27 @@ class Logger {
     std::vsnprintf(const_cast<char*>(output.c_str()), size, format, args);
     va_end(args);
 
-    printf("%s %s (%s) %s\n", Logger::LEVELS_STRINGS[l].c_str(), time.str().c_str(), _name.c_str(), output.c_str());
+    printf("%s %s (%s) %s %s\n",
+           Logger::LEVELS_STRINGS[l].c_str(),
+           time.str().c_str(),
+           _name.c_str(),
+           mdcToStr().c_str(),
+           output.c_str());
   }
+
+  void putMdc(const std::string& key, const std::string& val) { mdc_.emplace(key, val); }
+  void removeMdc(const std::string& key) { mdc_.erase(key); }
 };
 
 class Log {
  public:
   static Logger getLogger(std::string name) { return Logger(name); }
 };
-
+class MDC {
```

Comment:
Can you please add comments about what and MDC is, what it does and how it is used.

In the long run, we'll likely want to make it such that we can scope the MDC using RAII, rather than having put and remove macros all over the place. We'd have to get the scope right though for that to work, though.

Reply:
Done.
I changed the framework to support RAII.